### PR TITLE
Fix and clean up menu structure

### PR DESCRIPTION
**Pull request by marians**
This PR fixes the menus and does a few more things

- config.yaml contained entries for pages. I removed those for the principal menu and specified the according data in the pages's front matter.
- Several pages did not appear in the menu. I added them by adding the front matter data under the `menu:` key.
- The content of page `/overview/observability/logging/architecture/` moved up one level to `/overview/observability/logging/`, because there was no content on that node and the architecture page was the only one in the section.
- Adjusted some `linkTitle` to make the menu more readable

Before:

image

### CODE REVIEW

File: src/.hugo_build.lock
Comment (marians) [inline src/.hugo_build.lock:1]: This file shouldn't be in the repo

File: src/content/tutorials/registry/_index.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ---
 linkTitle: Registry
-title: Registry Guides
-description: Take a look at the list of guides that delve into configuring registries.
```
Comment (marians) [inline src/content/tutorials/registry/_index.md:4]: Removed the description, as there is only one tutorial in that section.

Comment (lyind) [reply src/content/tutorials/registry/_index.md:4]: grafik

How do we deal with the `description` being mandatory? Something like `description: ""`?

**Review by lyind — APPROVED** [review]
Not sure how to deal with empty descriptions without triggering the front-matter validator.

Otherwise LGTM

**Pull request merged**